doc/dataModel.md

```diff
@@ -0,0 +1,242 @@
+| type | String | Required | The document type for storing GPII app installation clients. | The value must be "gpiiAppInstallationClient". |
+| schemaVersion | String | Required | The version of the schema that the current document structure corresponds with. | None |
+| name | String | Required | The user provided client name.| None |
+| userId or gpiiKey | String | Optional | The id of the user or the GPII key who creates this client. The intention of this field is to identify the person who manages this client. It's still in question whether this field should record userId or gpiiKey. | null |
+| computerType | enum of ["public", "private", "shared by trusted parties"] | Required | Identify the type of the computer where this GPII app is installed. | None |
+| timestampCreated | Date | Required | The timestamp when the token is created. | now() |
+| timestampUpdated | Date | Optional | The timestamp when the token is updated. | null |
+
+### GPII App Installation Authorizations
+
+| Name | Type | Required or Optional | Description | Default |
+| --- | --- | --- | --- | --- |
+| id | String | Required | The ID of the authorization record. | None |
+| type | String | Required | The document type for storing GPII app installation authorizations. | The value must be "gpiiAppInstallationAuthorization". |
+| schemaVersion | String | Required | The version of the schema that the current document structure corresponds with. | None |
+| clientId | String | Required | The client id that this authorization is assigned to. | None |
+| gpiiKey | String | Required | The GPII key whose associated user preferences that this authorization is authorized to access. | None |
+| clientCredentialId | String | Required | The client credential id that is used to request this authorization. | None |
+| accessToken | String | Required | The access token used to retrieved the protected user preferences. | None |
+| revoked | Boolean | Required | Whether this authorization has been revoked. | false |
+| revokedReason | String | Optional | The revoked reason. | null |
+| timestampCreated | Date | Required | The timestamp when the token is created. | now() |
+| timestampRevoked | Date | Optional | The timestamp when the token is revoked. | null |
+| timestampExpires | Date | Required | The timestamp when this authorization expires. | None |
+
+### Users
+
+Refers to [fluid-express-user](https://github.com/fluid-project/fluid-express-user/blob/master/tests/data/users.json) for
+samples of user records.
+
+## Field Examples and Explanations
+
+### prefsSafes.preferences
+
+Object. Required.
+
+* Each preferences set in a preferences safe is keyed by an internally generated id. The default preferences set is keyed by a fixed string "gpii-default".
+* Each preferences set contains a field called "name" that is to save the user defined name for this preferences set.
+
+Example:
+
+```
+{
+    "flat": {
```

Comment:
The ontologies are still present in the `Prefs Safes/preferences` model, in these examples.  It is correct from a historical viewpoint, but my understanding is that ontologies are to be removed.  This raises a more general issue about the conflict between the goals of preserving the old documentation but also moving forward and documenting the data model that this general preferences service is going to use.  Perhaps the best place to do that is to expand the TODOs.

I've added a comment about ontologies to the TODO section below